src/main/java/com/ust/sourcecourse/configuration/service/DBDataSourceService.java

```diff
@@ -140,6 +142,10 @@ public List<DBTable> searchTablesByTag(String tag) {
 		if (StringUtils.isNotBlank(tag)) {
 			sourceTables = sourceTableRepository.retrieveByTag(tag.toLowerCase());
 		}
+
+		if (sourceTables.isEmpty()) {
+			throw new ResourceNotFoundException("No tables found with the given tag.");
```

Comment:
It should be No Content. 
ResponseStatusException(HttpStatus.NO_CONTENT

Reply:
resolved